### PR TITLE
Add ShopConstraintEventListener

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR adds an event listener to make available the ShopContraint as a controller argument
| Type?             | new feature
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | I think QA By dev as it is covered by unit tests, we can add more tests if needed

### CODE REVIEW

**Review by jolelievre — APPROVED** [review]
Thanks @FabienPapet

File: src/PrestaShopBundle/EventListener/ShopConstraintListener.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+
```
Comment (atomiix) [inline src/PrestaShopBundle/EventListener/ShopConstraintListener.php]:
```suggestion
declare(strict_types = 1);

```

**Review by zuk3975 — CHANGES_REQUESTED** [review]
One change that @atomiix mentioned and will be good to go

File: src/PrestaShopBundle/EventListener/ShopConstraintListener.php (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/EventListener/ShopConstraintListener.php]: Ah good call, and the license header is missing in both PHP files as well

**Review by jolelievre — CHANGES_REQUESTED** [review]

**Review by kpodemski — DISMISSED** [review]

File: src/PrestaShopBundle/EventListener/ShopConstraintListener.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
```
Comment (jolelievre) [inline src/PrestaShopBundle/EventListener/ShopConstraintListener.php]:
```suggestion
```

Comment (jolelievre) [reply src/PrestaShopBundle/EventListener/ShopConstraintListener.php]: Usually no blank lines before the header

File: tests/Unit/PrestaShopBundle/EventListener/ShopConstraintListenerTest.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
+<?php
+
```
Comment (jolelievre) [inline tests/Unit/PrestaShopBundle/EventListener/ShopConstraintListenerTest.php]:
```suggestion
```

**Review by zuk3975 — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]

**Review by atomiix — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by jolelievre** [comment]
Automatic tests running here https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/3548952483

**Comment by hibatallahAouadni** [comment]
Hello @FabienPapet 

> I think QA By dev as it is covered by unit tests, we can add more tests if needed

So, should I add the `Waiting for Dev` label? or could you please provide a way to test your PR?

Thanks!

**Comment by jolelievre** [comment]
I tested the PR, everything happens as expected The request attribute is correctly wether multishop mode is set or not it adapts appropriately CI is green so are the automatic tests so it's QA validated and mergeable

**Comment by jolelievre** [comment]
Thanks @FabienPapet

**Pull request merged**